src/d3a/d3a_core/sim_results/area_throughput_stats.py

```diff
@@ -0,0 +1,75 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from d3a.d3a_core.util import area_name_from_area_or_iaa_name,  round_floats_for_ui, \
+    add_or_create_key, create_subdict_or_update, area_sells_to_child, child_buys_from_area
+
+
+class AreaThroughputStats:
+    def __init__(self):
+        self.results = {}
+        self.results_redis = {}
+        self.exported_energy = {}
+        self.imported_energy = {}
+
+    def update(self, area):
+        self.update_results(area)
+
+    def _calc_results(self, area, baseline_value, energy_profile, direction_key):
+        if len(energy_profile.keys()) > 0:
+            # as this is mainly a frontend feature,
+            # the numbers are rounded for both local and redis results
+            out_dict = {direction_key: {
+                "peak_energy_kWh": round_floats_for_ui(max(energy_profile.values())),
+                "peak_percentage": round_floats_for_ui(max(energy_profile.values()) /
+                                                       baseline_value * 100)
+                                     }}
+            create_subdict_or_update(self.results, area.name, out_dict)
+            create_subdict_or_update(self.results_redis, area.uuid, out_dict)
+
+    def update_results(self, area):
+        baseline_import = area.baseline_peak_energy_import_kWh
+        baseline_export = area.baseline_peak_energy_export_kWh
+        if (baseline_import is not None and baseline_import > 0) or \
+                (baseline_export is not None and baseline_export > 0):
+            self.aggregate_exported_imported_energy(area)
+            if baseline_import is not None and baseline_import > 0:
+                self._calc_results(area, baseline_import, self.imported_energy, "import")
```

Comment:
I think you can save the `baseline_import > 0` check since its a numerator and a zero value wouldn't crash it

Reply:
Yes, it would. Devision by 0 is not defined.